### PR TITLE
Goog cleanup

### PR DESCRIPTION
**Pull request by gberaudo**
Use standard javascript instead.

### CODE REVIEW

File: src/abstractsynchronizer.js [hunk 1]
```diff
@@ -247,13 +246,17 @@ olcs.AbstractSynchronizer.prototype.listenForGroupChanges_ = function(group) {
...
+  }).bind(this);
 
   listenAddRemove();
 
   listenKeyArray.push(group.on('change:layers', function(e) {
     contentKeys.forEach(function(el) {
-      goog.array.remove(listenKeyArray, el);
+      var i = listenKeyArray.indexOf(el);
+      goog.asserts.assert(i >= 0);
+      if (i >= 0) {
```
Comment (elemoine) [inline src/abstractsynchronizer.js]: Strange that you use both an assertion and a `if`.

Comment (gberaudo) [inline src/abstractsynchronizer.js]: The reason is that `splice()` accepts negative argument, which is definitely not what we want.
Having both the assertion and the `if` allows to always have a correct behaviour and both catch the issue at debug time.

Comment (elemoine) [inline src/abstractsynchronizer.js]: I don't get it. Why isn't the assertion sufficient?

Comment (gberaudo) [inline src/abstractsynchronizer.js]: Assertions only work in debug mode.

Comment (gberaudo) [inline src/abstractsynchronizer.js]: I removed the assertion. @elemoine, is this PR fine with you know?

Comment (elemoine) [inline src/abstractsynchronizer.js]: I didn't suggest to remove the assertion. But we've spent too much time on this already :) Please merge.

Comment (gberaudo) [inline src/abstractsynchronizer.js]: For the record, by removing the assertion the code matches the one in `goog.array.remove()`.

**Comment by gberaudo** [comment]
Thank you for the review @elemoine.

**Pull request merged**